cli/cliutils/cliutils.go

```diff
@@ -876,6 +876,89 @@ func AgbotGet(urlSuffix, credentials string, goodHttpCodes []int, structure inte
 	return httpCode
 }
 
+// Runs a LIST to the agbot secure API and fills in the specified json structure. if the structure is just a string, fill in the raw json.
+// If the list of goodHttpCodes is non-empty and none match the actual http code, it will exit with an error; otherwise, the actual code is returned
+func AgbotList(urlSuffix, credentials string, goodHttpCodes []int, structure interface{}) (httpCode int) {
+
+	// get message printer
+	msgPrinter := i18n.GetMessagePrinter()
+
+	// check the agbot url
+	agbot_url := GetAgbotSecureAPIUrlBase()
+	if agbot_url == "" {
+		Fatal(HTTP_ERROR, msgPrinter.Sprintf("HZN_AGBOT_URL is not defined"))
+	}
+
+	// query the agbot secure api
+	url := agbot_url + "/" + urlSuffix
+	apiMsg := "LIST " + url
+
+	Verbose(apiMsg)
+
+	httpClient := GetHTTPClient(config.HTTPRequestTimeoutS)
+
+	resp := InvokeRestApi(httpClient, "LIST", url, credentials, nil, "Agbot", apiMsg)
+	if resp.Body != nil {
+		defer resp.Body.Close()
+	}
+
+	respBody := io.Reader(resp.Body)
+	if resp.Header.Get("Content-type") == "application/octet-stream" {
+		// Show progress of binary files downloading
+		msgPrinter.Print("Downloading object")
+		chunkNumber := 0
+		respBody = &progressReader{resp.Body, func(r int) {
+			chunkNumber++
+			if r > 0 {
```

Comment:
Why do we need this for LIST? The response will never be very big. I expected this function to look a  lot more like the GET.

Reply:
I just got rid of the progress-checking code; my code for AgbotGet uses ExchangeGet and there wasn't an existing ExchangeList so I took the code from ExchangeGet and switched out GET for LIST